### ISSUE TITLE
Digest resolution returns invalid digest

### ISSUE THREAD

**Issue by burdiyan** [google/go-containerregistry#271]
```
docker pull alpine:3.6
3.6: Pulling from library/alpine
Digest: sha256:de5701d6a3a36dc6a5db260d21be0422fd30dd2d158c1e048b34263e73205cb6
Status: Image is up to date for alpine:3.6
```

```
crane digest alpine:3.6
sha256:18959d9c6bb2acd6d07d837dc52e6467ab0f7282b36f12322c90f29be101e36d
```

Crane was built using commit `2f3e3e1a55fb70c98801f944d7b6249f8e8a3e20`.

Similar bug occurred in a tool similar to crane: https://github.com/genuinetools/reg/issues/131

**Comment by jonjohnsonjr** [comment]
I don't think it's an invalid digest, just a different one :)

The `library/` images are all manifest lists, which I haven't implemented quite yet: #119 

`docker pull` is outputting the manifest list digest, but `crane digest` is printing the default image's `amd64/linux` digest.

```
$ curl -H "Accept: application/vnd.docker.distribution.manifest.list.v2+json"  -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:library/alpine:pull" | jq -r .token)"  "https://index.docker.io/v2/library/alpine/manifests/3.6" | sha256sum
de5701d6a3a36dc6a5db260d21be0422fd30dd2d158c1e048b34263e73205cb6

$ curl -H "Accept: application/vnd.docker.distribution.manifest.v2+json"  -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:library/alpine:pull" | jq -r .token)"  "https://index.docker.io/v2/library/alpine/manifests/3.6" | sha256sum
18959d9c6bb2acd6d07d837dc52e6467ab0f7282b36f12322c90f29be101e36d
```

**Comment by burdiyan** [comment]
Oh, my bad. I actually haven't tried to do `docker pull` with the digest crane returned, assuming it will not work. But it actually works.
